contracts/modules/claims/Incidents.sol

```diff
@@ -0,0 +1,310 @@
+    address indexed underlyingToken
+  );
+
+  event IncidentAdded(
+    address indexed productId,
+    uint incidentDate,
+    uint priceBefore,
+    uint priceAfter
+  );
+
+  function setTokens(
+    address[] calldata _productIds,
+    address[] calldata _coveredTokens,
+    address[] calldata _underlyingTokens
+  ) external onlyGovernance {
+
+    require(
+      _productIds.length == _coveredTokens.length,
+      "Incidents: protocols and covered tokens lengths differ"
+    );
+
+    require(
+      _productIds.length == _underlyingTokens.length,
+      "Incidents: protocols and underyling tokens lengths differ"
+    );
+
+    for (uint i = 0; i < _productIds.length; i++) {
+      address id = _productIds[i];
+
+      require(coveredToken[id] == address(0), "");
+      require(underlyingToken[id] == address(0), "");
+
+      coveredToken[id] = _coveredTokens[i];
+      underlyingToken[id] = _underlyingTokens[i];
+      emit TokenSet(id, _coveredTokens[i], _underlyingTokens[i]);
+    }
+  }
+
+  function addIncident(
+    address productId,
+    uint incidentDate,
+    uint priceBefore,
+    uint priceAfter
+  ) external onlyGovernance {
+
+    require(priceBefore > priceAfter, "Incidents: No depeg");
+
+    address underlying = underlyingToken[productId];
+    require(underlying != address(0), "Incidents: Unsupported product");
+
+    Incident memory incident = Incident(
+      productId,
+      uint32(incidentDate),
+      uint112(priceBefore),
+      uint112(priceAfter)
+    );
+
+    incidents.push(incident);
+
+    emit IncidentAdded(productId, incidentDate, priceBefore, priceAfter);
+  }
+
+  function redeemPayout(
+    uint coverId,
+    uint incidentId,
+    uint coveredTokenAmount
+  ) external returns (uint claimId, uint payoutAmount) {
+
+    QuotationData qd = quotationData();
+    TokenController tc = tokenController();
+
+    Incident memory incident = incidents[incidentId];
+    address coverOwner;
+    uint sumAssured;
+    bytes4 currency;
+
+    {
+      address productId;
+      uint coverStartDate;
+      uint coverExpirationDate;
+
+      (
+      productId, coverOwner, coverStartDate,
+      coverExpirationDate, sumAssured, currency
+      ) = _getCoverDetails(qd, coverId);
+
+      // check ownership and covered protocol
+      require(msg.sender == coverOwner, "Incidents: Not cover owner");
+      require(productId == incident.productId, "Incidents: Bad incident id");
+
+      // check cover validity
+      require(coverStartDate <= incident.date, "Incidents: Cover start date is before the incident");
+      require(coverExpirationDate >= incident.date, "Incidents: Cover end date is after the incident");
+
+      // check grace period
+      uint gracePeriod = tc.claimSubmissionGracePeriod();
+      require(coverExpirationDate.add(gracePeriod) >= block.timestamp, "Incidents: Grace period has expired");
+    }
+
+    {
+      // assumes 18 decimals (eth & dai)
+      uint decimalPrecision = 1e18;
```

Comment:
why not const at the top ?

Reply:
because it will call `IERC20(asset).decimals()` at some point instead of being hardcoded so it will end up here